tests/Doctrine/Tests/Models/Rating/Business.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Models\Rating;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\GeneratedValue;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\JoinColumn;
+use Doctrine\ORM\Mapping\JoinTable;
+use Doctrine\ORM\Mapping\ManyToMany;
+use Doctrine\ORM\Mapping\Table;
+
+/**
+ * @Entity
+ * @Table(name="ratings_business")
+ */
+class Business
+{
+    /**
+     * @Id
+     * @Column(type="string")
+     * @GeneratedValue(strategy="NONE")
+     */
+    private string $id;
+
+    /**
+     * @psalm-var string|null
+     * @Column(type="string", length=255)
+     */
+    private string $name;
+
+    /**
+     * @psalm-doc https://www.doctrine-project.org/projects/doctrine-orm/en/2.16/reference/association-mapping.html#one-to-many-unidirectional-with-join-table
+     * @psalm-var Collection<Review>
+     * @ManyToMany(targetEntity="Review", cascade={"all"}, orphanRemoval=true, fetch="EAGER")
+     * @JoinTable(name="ratings_businesses_reviews",
+     *      joinColumns={@JoinColumn(name="business_id", referencedColumnName="id", onDelete="CASCADE")},
+     *      inverseJoinColumns={@JoinColumn(name="review_id", referencedColumnName="id", unique=true, onDelete="CASCADE")}
+     *      )
+     */
+    public $reviews;
+
+    public function __construct(string $id, string $name, ?Reviews $reviews = null)
+    {
+        $this->id   = $id;
+        $this->name = $name;
+        if ($reviews === null) {
+            $this->reviews = new ArrayCollection();
+        }
+    }
+
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    /** @psalm-return Collection<int, Review> */
+    public function getReviews(): Collection
+    {
+        return $this->reviews;
+    }
+
+    /**
+     * @return $this
+     */
+    public function setReviews(Reviews $reviews)
+    {
+        $this->reviews = $reviews;
```

Comment:
Please don't create a test fixture showing a case of replacing the whole collection. This is a very bad practice because it breaks the change tracking of Doctrine (as the collection is not the PersistentCollection anymore).

When modifying an entity loaded from the database, you must mutate the PersistentCollection that is in the property, not replacing it.

Reply:
In many cases I need to refresh an entity with its relations from outside the entity manages ie rebuilding it and then I need to save it. In the saving I need to know what to do as based on documentation my understanding was that a cascade operation with orphan removal would tell doctrine to do the reconciliation.

If this is not possible could you confirm in your opinion what the best approach would be? If i need to do some manual reconciliation do you have any suggestions on how to do it as it is not obvious to me that such changes would not be handled by the ORM